info-compiler/src/test/java/com/google/sps/WebCrawlerTest.java

```diff
@@ -59,13 +62,15 @@
   private final static String ABBREVIATED_CONTENT = ".";
   private final static int DELAY = 1;
 
-  private WebCrawler webCrawler;
-  private Datastore datastore;
+  private static WebCrawler webCrawler;
+  private static LocalDatastoreHelper datastoreHelper;
 
-  @Before
-  public void initialize() throws IOException {
-    this.datastore = mock(Datastore.class);
-    this.webCrawler = new WebCrawler(datastore);
+  @BeforeClass
+  public static void initialize() throws InterruptedException, IOException {
+    datastoreHelper = LocalDatastoreHelper.create();
+    datastoreHelper.start();
+    Datastore datastore = datastoreHelper.getOptions().getService();
+    webCrawler = new WebCrawler(datastore);
     NEWS_ARTICLE.setAbbreviatedContent(ABBREVIATED_CONTENT);
   }
```

Comment:
I think it's fine to keep these non-static and just create a new LocalDatastoreHelper for each test case (i.e. in `@Before` instead of `@BeforeClass`), and then you don't have to reset the datastore for each test.

Reply:
I just implemented this, but `datastoreHelper.stop()` sometimes generates a `java.net.ConnectException: Connection refused (Connection refused)`. This connection problem results from the Datastore emulator making HTTP requests. The number of times this problem occurs also varies between different runs of `WebCrawlerTest`. I believe it may be better to limit the number of times DatastoreHelper is created/destroyed to the minimum amount.

Reply:
Okay that makes sense.  In that case, would it make sense to move the "reset datastore" to a `@Before` method instead?

Reply:
Yeah definitely! That seems much better. I'll make the change